Changed default behavior: Activate progressbar only if running interactively

uellue opened the pull request:
Otherwise deactivate progress by default. Fix for issue #1842 

Currently dask uses stdout for progress bars. A fix to switch to stderr by default and give users control
is in preparation.

uellue committed:
Changed default behavior: Activate progressbar if stdout is a TTY, otherwise deactivate

Currently dask uses stdout for progress bars. A fix to switch to stderr by default and give users control
is in preparation.

francisco-dlp commented:
I am not sure if this fixes the issue because running from a script in a terminal still prints a progress bar. Only when the stdout of the script is redirected elsewhere the progress bar is not printed. This may sound like a good idea anyway but it can backfire. For example, we lose the progress bar in the jupyter notebook because the jupyter notebook is not a tty.

uellue commented:
@francisco-dlp funny that Jupyter doesn't emulate a terminal. The second option is
```
import __main__ as main
if hasattr(main, '__file__'):
    ...
```

uellue committed:
Instead of testing for TTY, test for the __file__ attribute of __main__ to determine if running interactively

Jupyter doesn't emulate a terminal, that means interactive use can only
be recognized through this method in a Jupyter environment.

thomasaarholt commented:
Just as a possiblity, one can check for iPython/Jupyter simply by `get_ipython` ([from here](https://stackoverflow.com/questions/44100477/how-to-check-if-you-are-in-a-jupyter-notebook)).

So something like this. When in the notebook or qtconsole, no import statement is necessary:
```python
try:
    get_ipython
    is_ipython = True
except:
    is_ipython = False

tjof2 commented:
Is this still an issue/a desired change?

tjof2 commented:
Replaced by #2548
